diff --git a/src/StreamWrapper/VideoRemoteStreamWrapper.php b/src/StreamWrapper/VideoRemoteStreamWrapper.php
--- a/src/StreamWrapper/VideoRemoteStreamWrapper.php
+++ b/src/StreamWrapper/VideoRemoteStreamWrapper.php
@@ -50,7 +50,7 @@
   /**
    * Returns the base path for stream wrapper.
    */
-  public static function basePath(\SplString $site_path = NULL) {
+  public static function basePath($site_path = NULL) {
     return static::baseUrl();
   }

